src/backup_domain.rs

```diff
@@ -22,3 +22,143 @@ use crate::pac::BKP;
+            41 => read_drx!(self, bkp_dr41),
+            42 => read_drx!(self, bkp_dr42),
+            x  => panic!("Backup Data Register dr{} does not exist.", x),
+        }
+    }
+
+    /// Write a 16-bit value to the Backup Data Register. The first argument
+    /// denotes the DRx register, i.e. 1 is DR1, up to DR42.
+    pub fn write_bkp_dr(&self, register: u8, data: u16) {
+        match register {
+            1  => write_drx!(self, dr1, data),
+            2  => write_drx!(self, dr2, data),
+            3  => write_drx!(self, dr3, data),
+            4  => write_drx!(self, dr4, data),
+            5  => write_drx!(self, dr5, data),
+            6  => write_drx!(self, dr6, data),
+            7  => write_drx!(self, dr7, data),
+            8  => write_drx!(self, dr8, data),
+            9  => write_drx!(self, dr9, data),
+            10 => write_drx!(self, dr10, data),
+            11 => write_drx!(self, bkp_dr11, data),
+            12 => write_drx!(self, bkp_dr12, data),
+            13 => write_drx!(self, bkp_dr13, data),
+            14 => write_drx!(self, bkp_dr14, data),
+            15 => write_drx!(self, bkp_dr15, data),
+            16 => write_drx!(self, bkp_dr16, data),
+            17 => write_drx!(self, bkp_dr17, data),
+            18 => write_drx!(self, bkp_dr18, data),
+            19 => write_drx!(self, bkp_dr19, data),
+            20 => write_drx!(self, bkp_dr20, data),
+            21 => write_drx!(self, bkp_dr21, data),
+            22 => write_drx!(self, bkp_dr22, data),
+            23 => write_drx!(self, bkp_dr23, data),
+            24 => write_drx!(self, bkp_dr24, data),
+            25 => write_drx!(self, bkp_dr25, data),
+            26 => write_drx!(self, bkp_dr26, data),
+            27 => write_drx!(self, bkp_dr27, data),
+            28 => write_drx!(self, bkp_dr28, data),
+            29 => write_drx!(self, bkp_dr29, data),
+            30 => write_drx!(self, bkp_dr30, data),
+            31 => write_drx!(self, bkp_dr31, data),
+            32 => write_drx!(self, bkp_dr32, data),
+            33 => write_drx!(self, bkp_dr33, data),
+            34 => write_drx!(self, bkp_dr34, data),
+            35 => write_drx!(self, bkp_dr35, data),
+            36 => write_drx!(self, bkp_dr36, data),
+            37 => write_drx!(self, bkp_dr37, data),
+            38 => write_drx!(self, bkp_dr38, data),
+            39 => write_drx!(self, bkp_dr39, data),
+            40 => write_drx!(self, bkp_dr40, data),
+            41 => write_drx!(self, bkp_dr41, data),
+            42 => write_drx!(self, bkp_dr42, data),
+            x  => panic!("Backup Data Register dr{} does not exist.", x),
+        }
+    }
+
+    /// Read an array of 42 elements (one element for every 16-bit DRx register)
+    /// from the Backup Data Register. The first element in the array (index 0)
+    /// is DR1.
+    pub fn read_bkp_dr_array(&self) -> [u16; 42] {
+        let mut data = [0; 42];
+        for i in 0..42 {
```

Comment:
This is not idiomatic as `clippy` will tell you:
```
   = note: #[warn(clippy::needless_range_loop)] on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_range_loop
```

Should be something like:
```
        for (i, d) in data.iter_mut().enumerate() {
            *d = self.read_bkp_dr((i + 1) as u8);
```
instead

Reply:
```rust
pub fn read_array(&self, buf: &mut [u16], start: usize) {
```
looks more useful then reading of full register block.

Reply:
You're right. I've changed it.